src/gg.regression.unity.bots/Runtime/Scripts/RGTestUtils.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections;
+using RegressionGames.StateRecorder;
+using UnityEngine;
+using UnityEngine.Assertions;
+using UnityEngine.SceneManagement;
+using Object = UnityEngine.Object;
+
+namespace RegressionGames
+{
+    /// <summary>
+    /// Utilities for running tests within the Unity Test Runner using Regression Games features.
+    /// </summary>
+    public class RGTestUtils
+    {
+        
+        /// <summary>
+        /// Waits until a specific scene has been loaded, and asserts that it has been loaded
+        /// </summary>
+        /// <param name="sceneName">The name of the scene to wait for</param>
+        /// <param name="timeout">The maximum time to wait for the scene to load (in seconds). Defaults to 5.</param>
+        public static IEnumerator WaitForScene(string sceneName, int timeout = 5)
+        {
+            var startTime = Time.time;
+            bool loaded = false;
+            while (!loaded)
+            {
+                var scene = SceneManager.GetSceneByName(sceneName);
+                loaded = scene.IsValid() && scene.isLoaded;
+                if (!loaded)
+                {
+                    var timeElapsed = Time.time - startTime;
+                    if (timeElapsed > timeout)
+                    {
+                        break;
+                    }
+                }
+                yield return null;
+            }
+            yield return null;
+            Assert.IsTrue(loaded);
```

Comment:
I would recommend adding a `message` parameter to this assertion to more clearly indicate to the developer the reason for the assertion failure (e.g. $"Scene {sceneName} failed to load within {timeout} seconds").